### ISSUE TITLE
newly proposed blocks should be marked "head", and commited should be marked "safe"

### ISSUE THREAD

**Issue by noot** [astriaorg/astria#50]
When tendermint proposes a new block, the block contains the commit for the parent block. However, the conductor currently treats this incorrectly, as when it receives a newly-proposed block, it executes it and marks it as safe/softly-committed, even though it's not actually finalized by tendermint yet. The correct logic when receiving a newly proposed block would be:
- execute and mark newly proposed block as "head" of chain
- verify commit contained in block (which is for the parent block), if it's verified, then mark parent block as "safe"
- receiving block from DA remains the same

Requires:
- updating the gRPC service to have methods such as:
```
ExecuteAndMarkAsHead(txs, prev_block_hash)
SetBlockAsSafe(block_hash)
SetBlockAsFinal(block_hash)
```

- updating the geth fork to support these methods
- updating conductor to correctly handle newly proposed vs committed blocks

blocked by #44

**Comment by noot** [comment]
yeah i think this is probably a duplicate of #207 at this point, will close

> there is no need to validate commitment to the rollup data 

we still need to do this when reading from DA
